src/halmos/sevm.py

```diff
@@ -1352,21 +1380,44 @@ def call(
         if not ret_size >= 0:
             raise ValueError(ret_size)
 
+        if arg_size > 0:
+            arg = wload(ex.st.memory, arg_loc, arg_size)
+        else:
+            arg = None
+
         caller = ex.prank.lookup(ex.this, to)
 
-        orig_code = ex.code.copy()
-        orig_storage = deepcopy(ex.storage)
-        orig_balance = ex.balance
-        orig_log = ex.log.copy()
+        def send_callvalue(condition=None) -> None:
+            if (
+                # no balance update for CALLCODE which transfers to itself
+                op == EVM.CALL
+                and not (is_bv_value(fund) and fund.as_long() == 0)
+            ):
+                # assume balance is enough; otherwise ignore this path
+                ex.solver.add(UGE(ex.balance_of(caller), fund))
```

Comment:
would it make sense to add this constraint to the `path` list or not really?

Reply:
`path` is only for debugging purposes, so it doesn't really matter, but we can add this.

Reply:
added